tests/Unit/App/Webhook/PdkWebhookManagerTest.php

```diff
@@ -21,7 +21,65 @@
 
 usesShared(new UsesMockPdkInstance(), new UsesMockEachCron());
 
-it('dispatches incoming webhook', function (string $hook) {
+it('dispatches and executes webhooks with myparcel header', function (string $hook) {
+    /** @var PdkWebhooksRepositoryInterface $repository */
+    $repository = Pdk::get(PdkWebhooksRepositoryInterface::class);
+    /** @var PdkWebhookManagerInterface $webhookManager */
+    $webhookManager = Pdk::get(PdkWebhookManagerInterface::class);
+    /** @var \MyParcelNL\Pdk\Tests\Bootstrap\MockCronService $cronService */
+    $cronService = Pdk::get(CronServiceInterface::class);
+
+    $repository->storeHashedUrl('https://example.com/hashed-url');
+
+    $repository->store(new WebhookSubscriptionCollection([['hook' => $hook, 'url' => $repository->getHashedUrl()]]));
+
+    $time = time();
+
+    $request = Request::create(
+        $repository->getHashedUrl(),
+        Request::METHOD_POST,
+        [],
+        [],
+        [],
+        ['HTTP_X_MYPARCEL_HOOK' => $hook],
+        json_encode([
+            'data' => [
+                'hooks' => [
+                    array_merge(['event' => $hook]),
+                ],
+            ],
+        ])
+    );
+
+    $response = $webhookManager->call($request);
+
+    $scheduled = $cronService->getScheduledTasks();
+    $timestamp = $scheduled->first()['timestamp'];
+
+    expect($response->getStatusCode())
+        ->toBe(Response::HTTP_ACCEPTED)
+        ->and($request->headers->all())
+        ->toHaveKey('x-myparcel-hook')
+        ->and($response->getContent())
+        ->toBe('')
+        ->and($response->getStatusCode())
+        ->toBe(Response::HTTP_ACCEPTED)
+        ->and($scheduled->all())
+        ->toHaveLength(1)
+        // The webhook is scheduled to start immediately. Add a little window to account for the time it takes to run the test.
+        ->and($timestamp)
+        ->toBeGreaterThanOrEqual($time - 10)
+        ->and($timestamp)
+        ->toBeLessThanOrEqual($time + 10);
+
+    $cronService->executeAllTasks();
```

Comment:
waarom run je ze als je niet kijkt naar het resultaat? nu ben je de cron service aan het testen. je kunt wel kijken of het uiteindelijk de juiste action aanroept